Source/Core/DolphinQt2/CheatsManager.cpp

```diff
@@ -0,0 +1,624 @@
+  {
+    m_match_length->addItem(option);
+  }
+
+  for (const auto& option : {tr("Equals to"), tr("Not equals to"), tr("Less than"),
+                             tr("Less or equal to"), tr("More than"), tr("More or equal to")})
+  {
+    m_match_operation->addItem(option);
+  }
+
+  auto* group_box = new QGroupBox(tr("Type"));
+  auto* group_layout = new QHBoxLayout;
+  group_box->setLayout(group_layout);
+
+  m_match_decimal = new QRadioButton(tr("Decimal"));
+  m_match_hexadecimal = new QRadioButton(tr("Hexadecimal"));
+  m_match_octal = new QRadioButton(tr("Octal"));
+
+  group_layout->addWidget(m_match_decimal);
+  group_layout->addWidget(m_match_hexadecimal);
+  group_layout->addWidget(m_match_octal);
+
+  layout->addWidget(m_result_label);
+  layout->addWidget(m_match_length);
+  layout->addWidget(m_match_operation);
+  layout->addWidget(m_match_value);
+  layout->addWidget(group_box);
+  layout->addWidget(m_match_new);
+  layout->addWidget(m_match_next);
+  layout->addWidget(m_match_refresh);
+  layout->addWidget(m_match_reset);
+
+  // Splitters
+  m_option_splitter = new QSplitter(Qt::Horizontal);
+  m_table_splitter = new QSplitter(Qt::Vertical);
+
+  m_table_splitter->addWidget(m_match_table);
+  m_table_splitter->addWidget(m_watch_table);
+
+  m_option_splitter->addWidget(m_table_splitter);
+  m_option_splitter->addWidget(options);
+
+  return m_option_splitter;
+}
+
+size_t CheatsManager::GetTypeSize() const
+{
+  switch (static_cast<DataType>(m_match_length->currentIndex()))
+  {
+  case DataType::Byte:
+    return sizeof(u8);
+  case DataType::Short:
+    return sizeof(u16);
+  case DataType::Int:
+    return sizeof(u32);
+  case DataType::Float:
+    return sizeof(float);
+  case DataType::Double:
+    return sizeof(double);
+  default:
+    return m_match_value->text().toStdString().size();
+  }
+}
+
+template <typename T>
+static bool Compare(T mem_value, T value, CompareType op)
+{
+  switch (op)
+  {
+  case CompareType::Equal:
+    return mem_value == value;
+  case CompareType::NotEqual:
+    return mem_value != value;
+  case CompareType::Less:
+    return mem_value < value;
+  case CompareType::LessEqual:
+    return mem_value <= mem_value;
+  case CompareType::More:
+    return value > mem_value;
+  case CompareType::MoreEqual:
+    return value >= mem_value;
+  default:
+    return false;
+  }
+}
+
+bool CheatsManager::MatchesSearch(u32 addr) const
+{
+  const auto text = m_match_value->text();
+  const auto op = static_cast<CompareType>(m_match_operation->currentIndex());
+
+  const int base =
+      (m_match_decimal->isChecked() ? 10 : (m_match_hexadecimal->isChecked() ? 16 : 8));
+
+  switch (static_cast<DataType>(m_match_length->currentIndex()))
+  {
+  case DataType::Byte:
+    return Compare<u8>(PowerPC::HostRead_U8(addr), text.toUShort(nullptr, base) & 0xFF, op);
+  case DataType::Short:
+    return Compare<short>(PowerPC::HostRead_U16(addr), text.toUShort(nullptr, base), op);
```

Comment:
You likely want `u16` here (or just omit it entirely, since it'll deduce the types), otherwise these are being compared as signed values instead of unsigned.